osu.Game/Screens/SelectV2/SongSelect.cs

```diff
@@ -585,6 +579,32 @@ private void newItemsPresented(IEnumerable<CarouselItem> carouselItems)
                 carousel.NextRandom();
         }
 
+        private IDisposable? noResultsDuck;
+
+        private void updateNoResultsPlaceholder(int count)
+        {
+            if (count == 0)
+            {
+                noResultsPlaceholder.Show();
+                noResultsPlaceholder.Filter = carousel.Criteria;
+
+                noResultsDuck ??= music.Duck(new DuckParameters { DuckVolumeTo = 1 });
+
+                // TODO: this is a slightly dodgy method to dim the background. usually you would control DimWhenUserSettingsIgnored,
+                // but using that applies an 800ms dim transform, and that feels too slow when combined with the duck.
+                ApplyToBackground(b => b.FadeColour(Color4.White.Opacity(0.5f), 500, Easing.OutQuint));
```

Comment:
You can't just do stuff like this. The testing overhead to make sure you haven't broken everything everywhere is immense.

Reply:
I tried that as well and it also didn't feel good.